### ISSUE TITLE
Useful string representation of networks

### ISSUE THREAD

**Issue by sylvlecl** [powsybl/pypowsybl#171]
* **Do you want to request a *feature* or report a *bug*?**

Feature

* **What is the current behavior?**

When printing a network, its representation is something like:
```python
print(network)
<pypowsybl.network.Network at 0x7fe73697a8d0>
```

* **What is the expected behavior?**

We could print more meaningful info, like the network name etc. See network metadata class.

* **What is the motivation / use case for changing the behavior?**

Being able to have useful information when printing a network, to help understand the data we're working on.

**Comment by yichen88** [comment]
#179

**Comment by sylvlecl** [comment]
Implemented in #186 , using the newly available network metadata from #180
